sdk/src/main/java/io/customer/sdk/queue/Queue.kt

```diff
@@ -0,0 +1,39 @@
+package io.customer.sdk.queue
+
+import io.customer.sdk.CustomerIOConfig
+import io.customer.sdk.queue.type.QueueModifyResult
+import io.customer.sdk.queue.type.QueueStatus
+import io.customer.sdk.util.JsonAdapter
+import io.customer.sdk.util.Logger
+
+class Queue internal constructor(
+    private val storage: QueueStorage,
+    private val jsonAdapter: JsonAdapter,
+    private val sdkConfig: CustomerIOConfig,
+    private val logger: Logger
+) {
+
+    fun <TaskData : Any> addTask(type: String, data: TaskData): QueueModifyResult {
```

Comment:
Should the `type` of a task be an `enum`? will it give us any benifit?

Reply:
When I do [this task](https://github.com/customerio/issues/issues/6674) you will understand why we have to use a `String`. In iOS, I used an enum until I did that task and I had to change it to using a String. To describe it briefly, you can't reference an `enum` that exists in the `messagingpush` module from the `tracking` module where the queue code exists. 

My plan is the same as how the iOS implementation is:

```kt
enum QueueTask {
  FOO

  val string: String {
    when (this) {
      FOO -> return "foo"
    }
  }
}

queue.addTask(QueueTask.FOO.string....)
```

So, the queue requires a String, but we still use an `enum` when calling `addTask()`.